app/models/user.rb

```diff
@@ -86,6 +86,10 @@ def github_import=(import)
     @github_import = import
   end
 
+  def visible_teams
```

Comment:
Personal taste, but I think it's overkill to a) introduce a new method in the `user` model for this and b) merge the two queries which roughly do the same...

How about refactoring the `visible` class method on the `Team` model into a scope? It's technically speaking already this... Then you'll be able to simply chain the scopes into something like
```ruby
user.teams.accepted.visible
```

What do you think?